cocos/platform/apple/CCFileUtils-apple.mm

```diff
@@ -52,18 +52,74 @@ void setBundle(NSBundle* bundle) {
     NSBundle* bundle_;
 };
 
+static id convertCCValueToNSObject(const cocos2d::Value &value);
+static cocos2d::Value convertNSObjectToCCValue(id object);
+
 static void addValueToDict(id nsKey, id nsValue, ValueMap& dict);
 static void addObjectToNSDict(const std::string& key, const Value& value, NSMutableDictionary *dict);
+static void addValueToArray(id item, ValueVector& array);
+static void addObjectToNSArray(const Value& value, NSMutableArray *array);
+
+static id convertCCValueToNSObject(const cocos2d::Value &value)
+{
+    switch (value.getType()) {
+        case Value::Type::NONE:
+            return [NSNull null];
+            
+        case Value::Type::STRING:
+            return [NSString stringWithCString:value.asString().c_str() encoding:NSUTF8StringEncoding];
+            
+        case Value::Type::BYTE:
+            return [NSNumber numberWithInt:value.asByte()];
+            
+        case Value::Type::INTEGER:
+            return [NSNumber numberWithInt:value.asInt()];
+            
+        case Value::Type::UNSIGNED:
+            return [NSNumber numberWithUnsignedInt:value.asUnsignedInt()];
+            
+        case Value::Type::FLOAT:
+            return [NSNumber numberWithFloat:value.asFloat()];
+            
+        case Value::Type::DOUBLE:
+            return [NSNumber numberWithDouble:value.asDouble()];
+            
+        case Value::Type::BOOLEAN:
+            return [NSNumber numberWithBool:value.asBool()];
+            
+        case Value::Type::VECTOR: {
+            NSMutableArray *array = [NSMutableArray array];
+            const ValueVector &vector = value.asValueVector();
+            for (const auto &e : vector) {
+                addObjectToNSArray(e, array);
+            }
+            return array;
+        }
+            
+        case Value::Type::MAP: {
+            NSMutableDictionary *dictionary = [NSMutableDictionary dictionary];
+            const ValueMap &map = value.asValueMap();
+            for (auto iter = map.begin(); iter != map.end(); ++iter) {
+                addObjectToNSDict(iter->first, iter->second, dictionary);
+            }
+            return dictionary;
+        }
+            
+        case Value::Type::INT_KEY_MAP:
+            break;
```

Comment:
No, currently there are no warnings, because all possible branches are handled. E.g. if i remove ```case Value::Type::INT_KEY_MAP``` then there will appear warning *Enumeration value 'INT_KEY_MAP' not handled in switch*.
 But if there would be ```default``` statement then that warning will not appear and we will not know that we should separately handle new ```Value::Type``` (if such would appear, for example something like ```Value::Type::SHORT```).
I think compiler warnings is better than log, as it can be seen immediately after Value::Type update. Though currently there are quite much warnings so it may be hidden...